hs/app/reachpc/Main.hs

```diff
@@ -0,0 +1,125 @@
+{- HLINT ignore "Use newtype instead of data" -}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Main (main) where
+
+-- import ReachPC.DirScan
+-- import ReachPC.Config (readOrCreateConfigToml, Config)
+
+-- used for stdio green threading / streaming stuff
+-- import Control.Concurrent (ThreadId, forkIO)
+-- import Control.Monad (forever)
+-- runStdinThread :: (String -> IO ()) -> IO ThreadId
+-- runStdinThread callback = forkIO $ forever $ getLine >>= callback . (<> "\n") 
+
+import System.Environment (getEnvironment, getArgs)
+import Data.List (isPrefixOf, intercalate)
+import Data.List.Extra (splitOn)
+import NeatInterpolation (text)
+import Data.Text (Text)
+import qualified Data.Text.IO as TIO
+import Control.Monad (forM_)
+import System.Exit (die)
+import Data.Maybe (fromMaybe)
+
+main :: IO ()
+main = do
+  -- Parse cli args
+  CliOptions {..} <- parseCliOptions
+  forM_ co_error die
+
+  -- Grab forwarded env vars
+  let forwardedEnvVars = fromMaybe [] co_envVars
+  let shouldForwardVar var = "REACH" `isPrefixOf` var || var `elem` forwardedEnvVars
+  environment <- filter (shouldForwardVar . fst) <$> getEnvironment
+  
+  -- Exec command
+  case fst co_command of
+    "internal help" -> TIO.putStrLn helpMessage
+    "version" -> version
+    "config" -> config
+    "auth" -> auth
+    "local-down" -> localDown
+    "local-install" -> localInstall
+    "local-up" -> localUp
+    _ -> execRemote environment co_command
+
+  return ()
+
+execRemote :: [(String, String)] -> (String, [String]) -> IO ()
+execRemote env (cmd, args) = do
+  putStrLn "Executed on remote:"
+  putStrLn $ cmd <> " " <> show args
+  print env
+
+helpMessage :: Text
+helpMessage = [text|
```

Comment:
Doesn't the CLI parser do this for us?

Reply:
I made this with opt-applicative and it was way longer and harder to follow. I found just writing it out was drastically nicer for such a simple application.